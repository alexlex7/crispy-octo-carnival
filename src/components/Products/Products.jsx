import React from 'react';
import Container from '../Container/Container';
import Carousel from '../Carousel/Carousel';
import CarouselItem from '../CarouselItem/CarouselItem';
import ProductItem from '../ProductItem/ProductItem';
import PropTypes from 'prop-types';
import styles from './Products.module.css';

export default function Products({ products, handleFavorite }) {
  return (
    <Container>
      <div className={styles.wrapper}>
        <h2 className={styles.title}>Our Popular Furniture</h2>
        <p className={styles.text}>
          All of our furniture uses the best materials and choices for our
          customers. All of our furniture uses the best materials and choices
          for our customers.
        </p>
        <Carousel>
          {products.map((item) => (
            <CarouselItem key={item.id}>
              <ProductItem item={item} handleFavorite={handleFavorite} />
            </CarouselItem>
          ))}
        </Carousel>
      </div>
    </Container>
  );
}

Products.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      imageURL: PropTypes.string.isRequired,
      isFavorite: PropTypes.bool.isRequired,
    }).isRequired
  ).isRequired,
  handleFavorite: PropTypes.func,
};
