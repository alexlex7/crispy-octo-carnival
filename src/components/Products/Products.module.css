.wrapper {
  width: 1200px;
  margin: 0 auto;
  padding-top: 140px;
  padding-bottom: 140px;
}

.title {
  margin-bottom: 20px;

  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 52px;

  color: #242424;
}

.text {
  margin-bottom: 80px;
  max-width: 820px;

  font-family: 'DM Sans';
  font-style: normal;
  font-weight: 400;
  font-size: 20px;
  line-height: 27px;

  color: #383738;
}
