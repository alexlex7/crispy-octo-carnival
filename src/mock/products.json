[
  {
    "id": "1",
    "name": "White Swan Chair",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/chair.jpg",
    "isFavorite": false
  },
  {
    "id": "2",
    "name": "White Swan Table",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/table.jpg",
    "isFavorite": false
  },
  {
    "id": "3",
    "name": "White Swan Sofa",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/sofa.jpg",
    "isFavorite": false
  },
  {
    "id": "4",
    "name": "White Swan Chair",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/chair.jpg",
    "isFavorite": false
  },
  {
    "id": "5",
    "name": "White Swan Table",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/table.jpg",
    "isFavorite": false
  },
  {
    "id": "6",
    "name": "White Swan Sofa",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/sofa.jpg",
    "isFavorite": false
  },
  {
    "id": "7",
    "name": "White Swan Chair",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/chair.jpg",
    "isFavorite": false
  },
  {
    "id": "8",
    "name": "White Swan Table",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/table.jpg",
    "isFavorite": false
  },
  {
    "id": "9",
    "name": "White Swan Sofa",
    "price": 40,
    "imageURL": "http://localhost:8080/images/furniture/sofa.jpg",
    "isFavorite": false
  }
]
