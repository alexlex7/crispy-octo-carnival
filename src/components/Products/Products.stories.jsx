import React from 'react';
import Products from './Products';
import products from '../../mock/products.json';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Products/Products',
  component: Products,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <Products {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  products,
};
