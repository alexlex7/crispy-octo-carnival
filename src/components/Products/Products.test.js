import { render, screen } from '@testing-library/react';
import Products from './Products';
import products from '../../mock/products.json';

test('render product item', () => {
  const { asFragment } = render(<Products products={products} />);

  const text = screen.getAllByText(`${products[0].name}`);
  expect(text[0]).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
