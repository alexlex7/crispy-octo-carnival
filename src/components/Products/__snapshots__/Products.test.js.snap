// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render product item 1`] = `
<DocumentFragment>
  <div
    class="container "
  >
    <div
      class="wrapper"
    >
      <h2
        class="title"
      >
        Our Popular Furniture
      </h2>
      <p
        class="text"
      >
        All of our furniture uses the best materials and choices for our customers. All of our furniture uses the best materials and choices for our customers.
      </p>
      <div
        class="wrapper"
      >
        <div
          class="carousel"
        >
          <div
            class="inner"
            style="transform: translateX(-0%);"
          >
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Chair"
                    src="http://localhost:8080/images/furniture/chair.jpg"
                    title="White Swan Chair"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Chair
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Table"
                    src="http://localhost:8080/images/furniture/table.jpg"
                    title="White Swan Table"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Table
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Sofa"
                    src="http://localhost:8080/images/furniture/sofa.jpg"
                    title="White Swan Sofa"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Sofa
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Chair"
                    src="http://localhost:8080/images/furniture/chair.jpg"
                    title="White Swan Chair"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Chair
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Table"
                    src="http://localhost:8080/images/furniture/table.jpg"
                    title="White Swan Table"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Table
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Sofa"
                    src="http://localhost:8080/images/furniture/sofa.jpg"
                    title="White Swan Sofa"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Sofa
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Chair"
                    src="http://localhost:8080/images/furniture/chair.jpg"
                    title="White Swan Chair"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Chair
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Table"
                    src="http://localhost:8080/images/furniture/table.jpg"
                    title="White Swan Table"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Table
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carouselItem"
              style="width: 380px;"
            >
              <div
                class="listItem"
              >
                <div
                  class="thumb"
                >
                  <img
                    alt="White Swan Sofa"
                    src="http://localhost:8080/images/furniture/sofa.jpg"
                    title="White Swan Sofa"
                  />
                </div>
                <div
                  class="content"
                >
                  <div
                    class="textWrapper"
                  >
                    <p
                      class="text"
                    >
                      White Swan Sofa
                    </p>
                    <div
                      class="btnWrapper"
                    >
                      <label
                        class="label"
                      >
                        <input
                          class="checkbox"
                          type="checkbox"
                        />
                        <svg
                          class=""
                        >
                          heart.svg
                        </svg>
                      </label>
                      <button
                        class="btn"
                        type="button"
                      >
                        <svg
                          class="icon"
                        >
                          share.svg
                        </svg>
                      </button>
                    </div>
                  </div>
                  <div
                    class="container"
                  >
                    <p
                      class="price"
                    >
                      $40
                    </p>
                    <button
                      class="btn"
                      type="button"
                    >
                      Buy Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="indicators"
        >
          <button
            class="btn"
            disabled=""
          >
            <svg
              class="icon"
            >
              prev.svg
            </svg>
          </button>
          <button
            class="btn"
          >
            <svg
              class="icon"
            >
              next.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
